docs/api-reference.md

```diff
@@ -3,25 +3,623 @@ id: api-reference
+| Function Name             | Description                                                                                |
+| ------------------------- | ------------------------------------------------------------------------------------------ |
+| [connect](#connect)       | Initializes a connection to a storage.                                                     |
+| [disconnect](#disconnect) | Closes the storage connection.                                                             |
+| [drop](#drop)             | Removes the Read Model's data from storage.                                                |
+| [dispose](#dispose)       | Forcefully disposes all unmanaged resources that Read Models served by this connector use. |
 
-For instance, to create a new list in the shopping list app:
+### connect
+
+Initializes a connection to a storage. An implementation should return a store object.
+
+#### Arguments
+
+| Argument Name | Description                                       |
+| ------------- | ------------------------------------------------- |
+| readModelName | A read model for which to establish a connection. |
+
+#### Example
+
+<!-- prettier-ignore-start -->
+
+[mdis]:# (../tests/custom-readmodel-sample/connector.js#connect)
+```js
+const connect = async readModelName => {
+  fs.writeFileSync(`${prefix}${readModelName}.lock`, true, { flag: 'wx' })
+  readModels.add(readModelName)
+  const store = {
+    get() {
+      return JSON.parse(String(fs.readFileSync(`${prefix}${readModelName}`)))
+    },
+    set(value) {
+      fs.writeFileSync(`${prefix}${readModelName}`, JSON.stringify(value))
+    }
+  }
+  return store
+}
+```
+
+<!-- prettier-ignore-end -->
+
+### disconnect
+
+Closes the storage connection.
+
+#### Arguments
+
+| Argument Name | Description                 |
+| ------------- | --------------------------- |
+| store         | A store object.             |
+| readModelName | A read model to disconnect. |
+
+#### Example
+
+<!-- prettier-ignore-start -->
+
+[mdis]:# (../tests/custom-readmodel-sample/connector.js#disconnect)
+```js
+const disconnect = async (store, readModelName) => {
+  safeUnlinkSync(`${prefix}${readModelName}.lock`)
+  readModels.delete(readModelName)
+}
+```
+
+<!-- prettier-ignore-end -->
+
+### drop
+
+Removes the Read Model's data from storage.
+
+#### Arguments
+
+| Argument Name | Description                        |
+| ------------- | ---------------------------------- |
+| store         | A store object.                    |
+| readModelName | A Read Model whose data to remove. |
+
+#### Example
+
+<!-- prettier-ignore-start -->
+
+[mdis]:# (../tests/custom-readmodel-sample/connector.js#drop)
+```js
+const drop = async (store, readModelName) => {
+  safeUnlinkSync(`${prefix}${readModelName}.lock`)
+  safeUnlinkSync(`${prefix}${readModelName}`)
+}
+```
+
+<!-- prettier-ignore-end -->
+
+### dispose
+
+Forcefully disposes all unmanaged resources that Read Models served by this connector use.
```

Comment:
- Why 'Forcefully'? 
- "that Read Models served by this connector use"?

Reply:
See my [other comment](https://github.com/reimagined/resolve/pull/1194#discussion_r316636506)